services/motion/builtin/move_request.go

```diff
@@ -518,8 +520,34 @@ func (ms *builtIn) newMoveOnGlobeRequest(
 		return nil, err
 	}
 
+	// convert obstacles of type []GeoGeometry into []Geometry
 	geomsRaw := spatialmath.GeoGeometriesToGeometries(obstacles, origin)
 
+	// convert bounding regions which are GeoGeometries into Geometries
+	boundingRegions := spatialmath.GeoGeometriesToGeometries(req.BoundingRegions, origin)
+
+	// get robot geometries
+	robotGeoms, err := b.Geometries(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(boundingRegions) > 0 {
+		// check that the robot geometries are within the bounding region
+		collisionCheck := motionplan.NewBoundingRegionConstraint(robotGeoms, boundingRegions, defaultCollisionBufferMM)
+		if !collisionCheck(&ik.State{}) {
+			return nil, fmt.Errorf("base named %s is not within the provided bounding regions", b.Name().ShortName())
+		}
+
+		// check that the dest is within the bounding region
+		collisionCheck = motionplan.NewBoundingRegionConstraint(
```

Comment:
IMO these checks should be within `motionplan`, not the motion service

Reply:
I'm fine with that. Thoughts on plopping in into the `Replan` function right at the top?

Reply:
Sounds great